### ISSUE TITLE
Different Font Size on Window and Heroku Machine

### ISSUE THREAD

**Issue by rudolfschmidt** [wkhtmltopdf/wkhtmltopdf#2171]
Hi,

I have installed wkhtmltopdf Version 0.12.2.1 with patched qt at a windows 8 machine and also at heroku with a buildpack. I want to use node for converting a html page to a pdf page at server side. Everything works fine, but the problem is, that the fonts are rendered different at heroku. The result at windows is fine, but at the heroku machine the fonts seems to be rendered bigger.
I post both examples, using the same server backend code for rendering.

![local](https://cloud.githubusercontent.com/assets/1162911/5910472/b4a844ae-a5ba-11e4-801e-d92daaeaaa3a.png)

![server](https://cloud.githubusercontent.com/assets/1162911/5910471/b4a6af40-a5ba-11e4-9d95-039974fd8305.png)

Is this a known bug or is there some workaround or another hint for a solution?

**Comment by RogerJoensson** [comment]
I wonder if this isn't the same problem as I am having: https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2156
(comment 10. There is also a small testfile)

For me (on Mac OS X) converting a page made for 96 dpi results in a slightly larger page as if it was processed as 92-93dpi. Even if I pass through a bitmap picture made for 96dpi (in the html file), it gets enlarged as if it were processed as 92-93dpi for output at 96.

If I pass a full A4 page made of a bitmap picture it gets smaller sometimes. It seems like it is automatically reduced in size to fit within the margins(?). Zeroing default margins with arguments -L 0 -R 0 -T 0 -B 0, brings it back in size, close to 96dpi. If the picture is smaller than a full page it gets slightly too large.

You differences seems bigger than the ones I get, but I still wonder if it is not the same problems?
What if you pass a bitmap picture smaller than full page?

**Comment by rudolfschmidt** [comment]
I found a dirty workaround with zoom. It seems that heroku uses 72dpi for rendering like mentioned here: https://github.com/wkhtmltopdf/wkhtmltopdf/issues/1782#issuecomment-46570110

but my Windows8 uses the default 96 dpi. You need to compensate by calculating 72 / 96= 0.75 and use zoom 0.75 if you want the same result like at a windows8 machine, because the default is zoom 1. 

But its not a nice solution because you are not os independent. maybe there is a fix in the future.

**Comment by wemersonjanuario** [comment]
any fix ?

**Comment by ashkulz** [comment]
This won't be fixed in the `0.12.x` series; it may be fixed by `0.13.0-beta` (as per the roadmap) -- there is no ETA when it will happen..

**Comment by DarkSide666** [comment]
And this is very bad that it won't / can't be fixed in 0.12.x versions :(

**Comment by pqr** [comment]
I would love to pay someone to fix this!
As workaround: I use dedicated Windows instance where I made simple web-service with wkhtmltopdf under the hood.

**Comment by fnicollet** [comment]
The workaround is quite simple, like @rudolfschmidt said, you need to use a specific zoom value for Linux environments. In Java I just did:
```
if (SystemUtils.IS_OS_LINUX){
	args.add("--zoom");
	args.add("0.78125");
}
```
args are my ProcessBuilder arguments and SystemUtils are from Apache Commons

Fabien

**Comment by DarkSide666** [comment]
Changing zoom factor depending on server OS works. It's quite simple workaround.

**Comment by punchi** [comment]
Fixed with zoom 0.8 for AWS Linux AMI (CentOS), developing from Windows 7 (dev), thanks =)

**Comment by callaginn** [comment]
So if I have a 27in Retina iMac, would I need to pass a dpi of 442? Basically using the number I found here: http://dpi.lv/

I discovered `--dpi 442 --zoom 1.625 --disable-smart-shrinking` works on that iMac after 131 tests, but still need to know _why_, so that I can calculate it for two more unique machines.

So my questions are:
- Am I using a correct value when setting the dpi?
- Since one of those machines is a Windows Install without a monitor, use 72 to calculate the --zoom value? That is, zoom = (72 / 442) * 10. Technically that's 1.6289592760181, so I had to fiddle with the numbers a bit. Am I doing this right...or is there a better method?
- And I'm assuming everyone is using the `--disable-smart-shrinking` setting?

**Comment by callaginn** [comment]
Nope. That formula didn't work on my non-Retina iMac. It rendered a PDF where all the fonts were about 140% larger than they were supposed to be.

So does anyone know of any way to calculate the correct wkhtmltopdf settings across different operating systems and screen DPIs?
